### DISCUSSION TITLE
Multiple choice shopping list

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#15011] [Archive]
It is one of those questions is it possible 😉
I would like to create a shopping list. As for the case of one item, there is no problem because I create a field in relation to another collection from where I display the name of the item and I have a second field where I enter the amount I need.

However, if I wanted to make a list in one record using the O2M lib M2M relation, I do not have how to specify the quantity for a specific item.

Only a repeater could handle it, but no relations can be set in it.

Therefore, the case of a list with the needed quantity in each record is impossible to achieve.

An idea: Now data from another collection is displayed in a specific field (relations M2M etc). It would be good to create a field that will display only certain information (from another collection) and another information in the next field, depending on the main choice, and assign a field with, for example, quantity ... after reflection, maybe a form with such fields would fully settle it. Will such a possibility exist in Directus?

**Comment by rijkvanzanten** [comment]
You should be able to use a many to many field for that, where you put the quantity on the junction table:

```
lists        lists_products         products
-----        --------------         --------
             id
id      <->  list
             product           <->  id
             quantity
```

That way, in the display template of the m2m interface on `lists`, you can show something like `{{ product.name }} ({{ quantity }})`
